### ISSUE TITLE
Schottky diodes values and Battery mAh value, please

### ISSUE THREAD

**Issue by ghost** [TheAustrian/Wemos-D1-Mini-BME280-Weather-Station#1]
Hi,

I want to also use your Wemos D1 Mini Weather solar station but what are the values : 
- Voltage and Ampers of the Schottky diodes
- mAh of the 18650 battery

Sincerely thanks.

Miguipda ;-)

**Comment by TheAustrian** [comment]
Hi,

I'm using 1A 20V 1N5817 as diodes, but any with a maximum voltage above the solar panel and capable of handling its short circuit current should be fine.

As for the battery I use a very old 2000 mAh one, so it's probably well below that capacity now. 
If you buy a battery you should consider buying one from a reputable producer like LG, Samsung, Sanyo, Sony, etc. offered by a reputable dealer; otherwise - a lot of the time - you get less capacity than advertised...

**Comment by jestani** [comment]
Hi,
I did everything as you write, but the table "weather" (mysql database) does not fill up data. Any suggestions??
Jack

**Comment by TheAustrian** [comment]
@jestani Hard to say where it went wrong - you might want to check 

a) if you can access the php-page manually with your browser (espdata.php) - it should write something like "Only for my ESP!"
b) Make sure you added the correct mySQL access data in the php files
c) Check in phpMyAdmin, if you set up the tables correctly
d) Make sure the ESP has an internet connection and that it can also access the NTP server, since it won't write to the database if it doesn't have a current timestamp
e) if that doesn't help, you might need to do some debugging with the ESP-code (i.e. have it write to the serial monitor, if it can connect or not)

Cheers, Alex

**Comment by jestani** [comment]
Thank you Alex for advice. I already have the first success. When I set up a base on a paid server (previous servers free) base came to life and be filled with data. However, it is the last problem. The data do not create charts. Chart.html them does not capture.
Regards
Jack

**Comment by TheAustrian** [comment]
Do you get any errors when you open the error console in the browser when you visit chart.html?
Did you add the right database information in datafetch.php?

Cheers, Alex

**Comment by jestani** [comment]
In chart.html errors:
CanvasJS Error: Chart Container with id "multiChart" was not found
canvasjs.js:2096 Uncaught TypeError: Cannot read property 'style' of undefined

code:
        this._container = typeof (this._containerId) === "string" ? document.getElementById(this._containerId) : this._containerId;

```
    if (!this._container) {
        if (window.console)
            window.console.log("CanvasJS Error: Chart Container with id \"" + this._containerId + "\" was not found");
        return;
    }

    this._container.innerHTML = "";
```

**Comment by TheAustrian** [comment]
You can delete the multichart part of the code without problem, I just left it in in case someone wants to try it.  So if you delete the following that should fix it. (I never had problems, though, so I left it in.)

```
// Chart with label + y and y2 values or multiSeries Chart
        var chart = new CanvasJS.Chart("multiChart",
        {
            title:{
                text: "Chart with y and y2 values"
            },      
            data: [
            {
                type: "line",
                dataPoints: dataPointsTemp
            },
            {               
                type: "line",
                dataPoints: dataPointsHumidity
            }           
            ]
        });
        chart.render();
```

**Comment by jestani** [comment]
Unfortunately, I still have no charts :( but I will fight;)
I am sending you scans of the base portion (weather) Is it correctly filled?
Thanks for the help Alex
Best regards
[https://photos.google.com/photo/AF1QipO1HAYTvQZoZmevn8HLIdYrjmqD6MrZNVcshOIw](url)
[https://photos.google.com/photo/AF1QipNLpZX8QIffhs1ngE_mDJ4dmzp-uPmHDY64vvdJ](url)

**Comment by TheAustrian** [comment]
I can't open the screen shots, unfortunately.
Cheers, Alex

**Comment by jestani** [comment]
I don't now why, but just copy link to browser:
https://photos.google.com/photo/AF1QipO1HAYTvQZoZmevn8HLIdYrjmqD6MrZNVcshOIw
https://photos.google.com/photo/AF1QipNLpZX8QIffhs1ngE_mDJ4dmzp-uPmHDY64vvdJ

**Comment by TheAustrian** [comment]
I'm getting a 404 error either way.

**Comment by jestani** [comment]
I hope that now will be OK :)

https://drive.google.com/open?id=0B15EqRdiuXMhcHliaE1IZzlEUEE
https://drive.google.com/open?id=0B15EqRdiuXMhVzJUX2IxVjFhNEE

**Comment by TheAustrian** [comment]
Yes, it was ok, though it seems you have an error with your table. The id should auto-increment, seems it's 0 for all of your values. Also the timestamps should be more consistent. Did you use bigint for it?

It should look like this:

![](http://t-iso.com/ss_table.tiff)

![](http://t-iso.com/ss_structure.tiff)

**Comment by TheAustrian** [comment]
I would also delete the old, incorrect data from the testing stage in the table (wrong timestamp, no real values) - that way you see if it's working after you made your changes.

**Comment by jestani** [comment]
Corrected, id (auto_increment), Id and timestamp are bigint (20) and .... :(
I do not know why so timestamp jumps. And those empty measurements?
Today I already have enough. Thank You for Your time!
Kind regards
Jack

**Comment by TheAustrian** [comment]
I can't say, either - you might want to debug with the ESP connected to serial and let it print what it sends to the server.

**Comment by jestani** [comment]
I do not pay attention, that the modem resets every 20 seconds.
I cleaned the memory in wemos d1 mini, but it did not work. However, other applications work well (web server, etc.). Soon I get 2 new mini wemos d1 and then see how it will work.

https://drive.google.com/open?id=0B15EqRdiuXMhd3hTcHRRT0dWVFk

**Comment by TheAustrian** [comment]
Alright, seems it sometimes is only able to send partial data so that's why you have problems with the database/half-empty datasets.

**Comment by jestani** [comment]
Alex, i have two questions yet: 
1. Do you use Wemos d1 mini or equivalent WiFi d1 mini? 
2. Can I connect the module d1 mini directly to the Li-ion 18650?

**Comment by TheAustrian** [comment]
Jack, to answer your questions:
1) I use the original Wemos D1 Mini.
2) Yes, to the 5V pin. It is connected to a low dropout linear regulator, so you can supply anything between about 3.3V and 5V to that pin (maybe even slightly more, but I never tried). 
Cheers, Alex

**Comment by jestani** [comment]
Hi Alex,
When I removed the zeroing buffers, after some time the table began to fill properly. He stopped a reset.
What do You think about?

`String SensorData () {
  int i = 0;
  int j = 0;
  String textToSend ="";
  while ( i <= buffersize && j < maxsendsize) {
    if (timeBuffer[i] > 0) {
      if (textToSend != "") 
      //textToSend += "&";
          textToSend += "&d[" + String(i) + "][ti]=" + timeBuffer[i] + "&d[" + String(i) + "][t]=" + tempBuffer[i] + "&d[" + String(i) + "][p]=" + pressureBuffer[i] + "&d[" + String(i) + "][h]=" + humidityBuffer[i] + "&d[" + String(i) + "][v]=" + voltageBuffer[i];   
          j++;
    /*      timeBuffer[i] = 0;
          tempBuffer[i] = 0.0;
          pressureBuffer[i] = 0.0;
          humidityBuffer[i] = 0.0; 
          voltageBuffer[i] = 0.0;  */
          Serial.println(i); 
       }
    i++;  
  }
  return textToSend;
}`

**Comment by jestani** [comment]
Hi Alex, when I removed buffers zeroing in function (sensor data)
He stopped reset and after a time, the table was filled correctly.
What do you think?

String SensorData () {
  int i = 0;
  int j = 0;
  String textToSend ="";
  while ( i <= buffersize && j < maxsendsize) {
    if (timeBuffer[i] > 0) {
      if (textToSend != "") 
      //textToSend += "&";
          textToSend += "&d[" + String(i) + "][ti]=" + timeBuffer[i] + "&d[" + String(i) + "][t]=" + tempBuffer[i] + "&d[" + String(i) + "][p]=" + pressureBuffer[i] + "&d[" + String(i) + "][h]=" + humidityBuffer[i] + "&d[" + String(i) + "][v]=" + voltageBuffer[i];  
          j++;
       /\*   timeBuffer[i] = 0;
          tempBuffer[i] = 0.0;
          pressureBuffer[i] = 0.0;
          humidityBuffer[i] = 0.0; 
          voltageBuffer[i] = 0.0;  */
          Serial.println(i); 
       }
    i++;  
  }
  return textToSend;
}
